### DISCUSSION TITLE
What are the differences between the exogenous quantities distributions of oneshot and the one of std/col?

### DISCUSSION THREAD

**Discussion by rezunli96** [yasserfarouk/scml#15] [one-shot-qa] [answered]
Hello, I am watching the online [tutorial](https://www.youtube.com/watch?v=cEZ_opQ4fMY) which said the exogenous quantities distributions are "more consistent" in oneshot track. 

I am reading /oneshot/world.py and /scml2020/world.py where in generate() I believe lies on the relevant code. But is there any difference in the code regarding the parameters and the way it generate?

**Comment by yasserfarouk** [comment, marked as answer]
Actually there is no difference in the code now. Thre relevant code is in ```generate``` as you suggested:
```python
        # - divide the quantity at every level between factories
        exogenous_supplies = distribute_quantities(
            equal_exogenous_supply,
            exogenous_supply_predictability,
            quantities[0],
            n_agents_per_process[0],
            n_steps,
        )
        exogenous_sales = distribute_quantities(
            equal_exogenous_sales,
            exogenous_sales_predictability,
            quantities[-1],
            n_agents_per_process[-1],
            n_steps,
        )
```

Nevertheless, we **GUARNTEE** this in the oneshot game but may change it in the **standard** game. The main reason is to avoid having special treatment for the agents at second and second to last agents. Note that in the standard game, the depth can be anything. Say that we have a production graph of depth 5. In this case, only agents at level 0 and 4 will have exogenous contracts. If we guarantee consistent distribution, agents at level 1 nd 3 should take it into account but not agents at 2. To avoid nudging people into working on these special cases, we prefer NOT to guarantee consistent assignment anywhere.

Does this make sense?

**Reply by rezunli96** [reply]
Thanks for the reply! It makes perfectly sense.
